src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +203,1114 @@ const ActorInit En_Test3_InitVars = {
+            this->unk_D80 = 4;
+            func_80A40230(this, play);
+            Math_Vec3f_Copy(&this->player.actor.world.pos, &worldPos);
+            D_80A41D48 = false;
+            this->unk_D84 = 0.0f;
+        }
+    } else {
+        D_80A41D40 = 0.0f;
+        D_80A41D44 = this->player.actor.shape.rot.y;
+        this->unk_D94(this, play);
+        sInput.press.button = (sInput.rel.button ^ sInput.cur.button) & sInput.cur.button;
+        func_800B6F20(play, &sInput, D_80A41D40, D_80A41D44);
+    }
+    play->playerUpdate(&this->player, play, &sInput);
+    if (D_80A41D48) {
+        this->player.actor.world.pos.x = D_80A41D50.x;
+        this->player.actor.world.pos.z = D_80A41D50.z;
+        this->player.linearVelocity = 0.0f;
+    }
+}
+
+s32 D_80A418C8 = false;
+
+s32 EnTest3_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnTest3* this = THIS;
+
+    if (limbIndex == OBJECT_TEST3_LIMB_01) {
+        D_80A41D6C = &this->player.meleeWeaponInfo[2].base;
+        if (!(this->player.skelAnime.moveFlags & ANIM_FLAG_4) || (this->player.skelAnime.moveFlags & ANIM_FLAG_1)) {
+            pos->x *= this->player.ageProperties->unk_08;
+            pos->z *= this->player.ageProperties->unk_08;
+        }
+        if (!(this->player.skelAnime.moveFlags & ANIM_FLAG_4) ||
+            (this->player.skelAnime.moveFlags & ANIM_FLAG_UPDATEY)) {
+            pos->y *= this->player.ageProperties->unk_08;
+        }
+        pos->y -= this->player.unk_AB8;
+        if (this->player.unk_AAA != 0) {
+
+            Matrix_Translate(pos->x, ((Math_CosS(this->player.unk_AAA) - 1.0f) * 200.0f) + pos->y, pos->z,
+                             MTXMODE_APPLY);
+            Matrix_RotateXS(this->player.unk_AAA, MTXMODE_APPLY);
+            Matrix_RotateZYX(rot->x, rot->y, rot->z, MTXMODE_APPLY);
+            func_80125318(pos, rot);
+        }
+    } else {
+        if (*dList != NULL) {
+            D_80A41D6C++;
+        }
+        if (D_80A418C8) {
+            *dList = NULL;
+        }
+        if (limbIndex == OBJECT_TEST3_LIMB_0B) {
+            rot->x += this->player.unk_AAC.z;
+            rot->y -= this->player.unk_AAC.y;
+            rot->z += this->player.unk_AAC.x;
+        } else if (limbIndex == OBJECT_TEST3_LIMB_0A) {
+            s32 pad; // Must exist here to match
+
+            if (this->player.unk_AA8 != 0) {
+                Matrix_RotateZS(0x44C, MTXMODE_APPLY);
+                Matrix_RotateYS(this->player.unk_AA8, MTXMODE_APPLY);
+            }
+            if (this->player.unk_AB2.y != 0) {
+                Matrix_RotateYS(this->player.unk_AB2.y, MTXMODE_APPLY);
+            }
+            Matrix_RotateXS(this->player.unk_AB2.x, MTXMODE_APPLY);
+            if (this->player.unk_AB2.z != 0) {
+                Matrix_RotateZS(this->player.unk_AB2.z, MTXMODE_APPLY);
+            }
+        } else {
+            func_80125500(play, &this->player, limbIndex, pos, rot);
```

Comment:
Hmm, well we're going to have to check this one for recasting temps later anyway.